### PR TITLE
feat: mime types on tools, show upload btn on demand

### PR DESCRIPTION
**Pull request by Saghen**
Adds a mimetypes property on the tools used by the frontend to control the `UploadBtn`'s `accept` property. If there are no active mime types, the upload button will not be shown. Adds an enable/disable all button to the tools menu

image

### CODE REVIEW

File: src/lib/components/ToolsMenu.svelte [hunk 1]
```diff
@@ -50,6 +56,10 @@
 					>
 				{/if}
 			</div>
+			<div class="flex space-x-4 text-sm text-gray-500">
+				<button on:click={() => setAllTools(true)}>Enable all</button>
+				<button on:click={() => setAllTools(false)}>Disable all</button>
+			</div>
```
Comment (mishig25) [inline src/lib/components/ToolsMenu.svelte:62]:
```suggestion
			<div class="flex space-x-4 text-sm text-gray-500">
				<button
					disabled={allToolsEnabled}
					class:opacity-50={allToolsEnabled}
					on:click={() => setAllTools(true)}>Enable all</button
				>
				<button
					disabled={allToolsDisabled}
					class:opacity-50={allToolsDisabled}
					on:click={() => setAllTools(false)}>Disable all</button
				>
			</div>
```

and define after `$: activeToolCount` definition
```ts
	$: allToolsEnabled = activeToolCount === $page.data.tools.length;
	$: allToolsDisabled = activeToolCount === 0;
```

for a better UX

File: src/lib/components/UploadBtn.svelte [hunk 2]
```diff
@@ -3,6 +3,7 @@
 
 	export let classNames = "";
 	export let files: File[];
+	export let mimeTypes: string[] = ["*/*"];
```
Comment (mishig25) [inline src/lib/components/UploadBtn.svelte:6]:
```suggestion
	export let mimeTypes: string[];
```

`mimeTypes` prop always has valid value. No ?

File: src/lib/components/ToolsMenu.svelte (hunk 1 above)
Comment (gary149) [reply src/lib/components/ToolsMenu.svelte:62]: didn't try but I guess it should use <input type="checkbox"> with indeterminate state
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#indeterminate_state_checkboxes

Comment (Saghen) [reply src/lib/components/ToolsMenu.svelte:62]: image
image

thanks Mishig, looks great imo!

**Review by mishig25 — APPROVED** [review]
lgtm !

File: src/lib/components/ToolsMenu.svelte (hunk 1 above)
Comment (gary149) [reply src/lib/components/ToolsMenu.svelte]: could be smaller I can tweak it a bit 👀

**Comment by nsarrazin** [comment]
Works great! I extended support to the dropzone & onpaste handler in [this PR](https://github.com/huggingface/chat-ui/pull/1222)

**Review by gary149 — APPROVED** [review]

**Pull request merged**